src/libraries/Microsoft.PowerFx.Interpreter/Functions/CommonErrors.cs

```diff
@@ -0,0 +1,220 @@
+        }
+
+        // Invalid operation: division by zero
+        public static ErrorValue DivByZeroError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrDivByZero, ErrorKind.Div0, irContext.SourceContext));
+        }
+
+        public static ErrorValue OverflowError(IRContext irContext)
+        {
+            return PrimitiveValueConversions.OverflowError(irContext);
+        }
+
+        // Language code '{0}' not supported.
+        public static ErrorValue BadLanguageCode(IRContext irContext, string languageCode)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrBadLanguageCode, ErrorKind.BadLanguageCode, irContext.SourceContext, languageCode));
+        }
+
+        // The Date/Time could not be parsed.
+        public static ErrorValue InvalidDateTimeParsingError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrInvalidDateTimeParsingError, ErrorKind.InvalidArgument, irContext.SourceContext));
+        }
+
+        // Invalid date/time value.
+        public static ErrorValue InvalidDateTimeError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrInvalidDateTimeError, ErrorKind.InvalidArgument, irContext.SourceContext));
+        }
+
+        public static ErrorValue InvalidArgumentError(IRContext irContext, ErrorResourceKey errorResourceKey)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(errorResourceKey, ErrorKind.InvalidArgument, irContext.SourceContext));
+        }
+
+        // The value could not be interpreted as a Boolean.
+        public static ErrorValue InvalidBooleanFormatError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrInvalidBooleanFormatError, ErrorKind.InvalidArgument, irContext.SourceContext));
+        }
+
+        // The value could not be interpreted as a Color.
+        public static ErrorValue InvalidColorFormatError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrInvalidColorFormatError, ErrorKind.InvalidArgument, irContext.SourceContext));
+        }
+
+        // Unknown error.
+        public static ErrorValue UnreachableCodeError(IRContext irContext)
+        {
+            return new ErrorValue(
+                irContext,
+                NewExpressionError(RuntimeStringResources.ErrInvalidColorFormatError, ErrorKind.Validation, irContext.SourceContext));
```

Comment:
>Validation [](http://example.com/codeflow?start=96&length=10)

If 'UnreachableCode' is something that should never happen, this should be ErrorKind.Internal